master/internal/trial.go

```diff
@@ -755,6 +762,12 @@ func (t *trial) processLog(ctx *actor.Context, msg scheduler.ContainerLog) {
 }
 
 func (t *trial) processTaskTerminated(ctx *actor.Context, msg scheduler.TaskTerminated) {
+	if t.timedOut {
+		ctx.Log().Info("trial terminated after timing out")
+		ctx.Self().Stop()
```

Comment:
Could you just set `t.task = nil` and `return` here?  Or use `ctx.Self().Stop()` similarly when we detect that we've reached the maximum number of restarts?  Just so that they are treated the same.  I'm not sure which is better.

Reply:
I'll do `t.task = nil` and let the generic stopping logic handle it.